compiler/src/dotty/tools/dotc/core/Denotations.scala

```diff
@@ -365,31 +366,32 @@ object Denotations {
      *
      *  NoDenotations are dropped. MultiDenotations are handled by merging
      *  parts with same signatures. SingleDenotations with equal signatures
-     *  are joined as follows:
+     *  are joined by following this sequence of steps:
      *
-     *  In a first step, consider only those denotations which have symbols
-     *  that are accessible from prefix `pre`.
+     *  1. If exactly one the denotations has an inaccessible symbol, pick the other one.
+     *  2. Otherwise, if one of the infos overrides the other one, and the associated
+     *     symbol does not score strictly lower than the other one,
+     *     pick the associated denotation.
+     *  3. Otherwise, if the two infos can be combined with `infoMeet`, pick that as
+     *     result info, and pick the symbol that scores higher as result symbol,
+     *     or pick `sym2` as a tie breaker. The picked info and symbol are combined
```

Comment:
Why sym2 and not sym1 as the tie breaker?

Reply:
It was `sym2` before. I now went through the callers, but I can see no reason why it should be. Single denotations are merged with the left operand first in linearization order. So it makes some sense to keep it that way (maybe?, or it makes no difference at all, I am not sure).